File: e-livraria-web/src/models/DeliveryAddress.ts
import { ETypeResidence } from './ETypeResidence';

export interface IDeliveryAddressModel{
    id?: number;
    typeresidence?: ETypeResidence;
    patioType?: string;
    publicArea?: string;
    numberAddrs?: string;
    neighborhood?: string;
    cep?: string;
    city?: string;
    state?: string;
    country?: string;
    observations?: string;
    nameAddrs?: string;
}

export class DeliveryAddress {
    id: number;
    typeresidence: ETypeResidence;
    patioType: string;
    publicArea: string;
    numberAddrs: string;
    neighborhood: string;
    cep: string;
    city: string;
    state: string;
    country: string;
    observations: string;
    nameAddrs: string;

    constructor({
        id = 0,
        typeresidence = ETypeResidence.HOUSE,
        patioType = '',
        publicArea = '',
        numberAddrs = '',
        neighborhood = '',
        cep = '',
        city = '',
        state = '',
        country = '',
        observations = '',
        nameAddrs = '',
    }:IDeliveryAddressModel){
        this.id = id;
        this.typeresidence = typeresidence;
        this.patioType = patioType;
        this.publicArea = publicArea;
        this.numberAddrs = numberAddrs;
        this.neighborhood = neighborhood;
        this.cep = cep;
        this.city = city;
        this.state = state;
        this.country = country;
        this.observations = observations;
        this.nameAddrs = nameAddrs;
    }

    get getId(): number {
        return this.id;
    }
    set setId(id: number) {
        this.id = id;
    }
    get getTyperesidence(): ETypeResidence {
        return this.typeresidence;
    }
    set setTyperesidence(typeresidence: ETypeResidence) {
        this.typeresidence = typeresidence;
    }
    get getPatioType(): string {
        return this.patioType;
    }
    set setPatioType(patioType: string) {
        this.patioType = patioType;
    }
    get getPublicArea(): string {
        return this.publicArea;
    }
    set setPublicArea(publicArea: string) {
        this.publicArea = publicArea;
    }
    get getNumberAddrs(): string {
        return this.numberAddrs;
    }
    set setNumberAddrs(numberAddrs: string) {
        this.numberAddrs = numberAddrs;
    }
    get getNeighborhood(): string {
        return this.neighborhood;
    }
    set setNeighborhood(neighborhood: string) {
        this.neighborhood = neighborhood;
    }
    get getCep(): string {
        return this.cep;
    }
    set setCep(cep: string) {
        this.cep = cep;
    }
    get getCity(): string {
        return this.city;
    }
    set setCity(city: string) {
        this.city = city;
    }
    get getState(): string {
        return this.state;
    }
    set setState(state: string) {
        this.state = state;
    }
    get getCountry(): string {
        return this.country;
    }
    set setCountry(country: string) {
        this.country = country;
    }
    get getObservations(): string {
        return this.observations;
    }
    set setObservations(observations: string) {
        this.observations = observations;
    }

    get getNameAddrs(): string {
        return this.nameAddrs;
    }
    set setNameAddrs(nameAddrs: string) {
        this.nameAddrs = nameAddrs;
    }
}

File: ecommerceLivraria/src/main/java/com/nrabello/ecommerceLivraria/model/DeliveryAddress.java
package com.nrabello.ecommerceLivraria.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

@Entity
@Table(name="EnderecosEntrega")
public class DeliveryAddress {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="ede_id")
    private Long id;
    @Column(name="ede_tipo_residencia")
    @Enumerated(EnumType.STRING)
    private ETyperesidence typeresidence;
    @Column(name="ede_tipo_logradouro")
    private String patioType;
    @Column(name="ede_logradouro")
    private String publicArea;
    @Column(name="ede_numero")
    private String numberAddrs;
    @Column(name="ede_bairro")
    private String neighborhood;
    @Column(name="ede_cep")
    private String CEP;
    @Column(name="ede_cidade")
    private String city;
    @Column(name="ede_estado")
    private String state;
    @Column(name="ede_pais")
    private String country;
    @Column(name="ede_observacoes")
    private String observations;
    @Column(name = "ede_nome")
    private String nameAddrs;

    @ManyToOne
    @JoinColumn(name="ede_cli_id", referencedColumnName = "cli_id")
    @JsonIgnore
    private Client client;

    public DeliveryAddress() {
    }

    public DeliveryAddress(Long id, ETyperesidence typeresidence, String patioType, String publicArea, String numberAddrs, String neighborhood, String CEP, String city, String state, String country, String observations, Client client, String nameAddrs) {
        this.setId(id);
        this.setTyperesidence(typeresidence);
        this.setPatioType(patioType);
        this.setPublicArea(publicArea);
        this.setNumberAddrs(numberAddrs);
        this.setNeighborhood(neighborhood);
        this.setCEP(CEP);
        this.setCity(city);
        this.setState(state);
        this.setCountry(country);
        this.setObservations(observations);
        this.setClient(client);
        this.setNameAddrs(nameAddrs);
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }
    public ETyperesidence getTyperesidence() {
        return typeresidence;
    }

    public void setTyperesidence(ETyperesidence typeresidence) {
        this.typeresidence = typeresidence;
    }

    public String getPatioType() {
        return patioType;
    }

    public void setPatioType(String patioType) {
        this.patioType = patioType;
    }

    public String getPublicArea() {
        return publicArea;
    }

    public void setPublicArea(String publicArea) {
        this.publicArea = publicArea;
    }

    public String getNumberAddrs() {
        return numberAddrs;
    }

    public void setNumberAddrs(String numberAddrs) {
        this.numberAddrs = numberAddrs;
    }

    public String getNeighborhood() {
        return neighborhood;
    }

    public void setNeighborhood(String neighborhood) {
        this.neighborhood = neighborhood;
    }

    public String getCEP() {
        return CEP;
    }

    public void setCEP(String CEP) {
        this.CEP = CEP;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getObservations() {
        return observations;
    }

    public void setObservations(String observations) {
        this.observations = observations;
    }

    public Client getClient() {
        return client;
    }

    public void setClient(Client client) {
        this.client = client;
    }

    public String getNameAddrs() {
        return nameAddrs;
    }

    public void setNameAddrs(String nameAddrs) {
        this.nameAddrs = nameAddrs;
    }

}
